pandas/core/generic.py

```diff
@@ -3708,7 +3708,9 @@ class   animal   locomotion
             return result
 
         if axis == 1:
-            return self[key]
+            if drop_level:
+                return self[key]
+            return self[[key]]
```

Comment:
PITA but this causes us to make a copy instead of a view.  can we re-use any of the code in the rest of the method to keep this consistent?

Reply:
Would make this pretty complex probably, but will have a look tomorrow

Reply:
This should work, but got a bit more complex. Using iloc now to select the data if ``droplevel=False``